### DISCUSSION TITLE
Error: Invalid API Key Please select a .p8 private key

### DISCUSSION THREAD

**Discussion by lukas-h** [codemagic-ci-cd/codemagic-docs#1751] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Go to Team integrations
2. Go to (Apple) Developer Portal
3. In the dialog Apple Developer Portal integration, fill out everything
4. Try to upload the .p8 file

### Expected results

Should upload the .p8 file without error.

### Actual results

Does not upload the file. Instead I get this error: 
> Invalid API Key
Please select a .p8 private key
<img width="1017" alt="Screenshot 2023-06-09 at 14 20 55" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/10634693/7d490a18-9daf-4de3-ad34-7bece93400e6">

**Comment by himesh-cm** [comment, marked as answer]
Discussion continued on intercom.
I will update the thread once we have a resolution.

**Reply by himesh-cm** [reply]
`.p8` key has additional white space characters, and had to remove those and re-upload it.
